python/mcap/tests/test_reader.py

```diff
@@ -37,7 +41,7 @@ def test_make_not_seeking(pipe: Tuple[IO[bytes], IO[bytes]]):
 
 
 @pytest.mark.parametrize("reader_cls", [SeekingReader, NonSeekingReader])
-def test_all_messages(reader_cls: Type[McapReader]):
+def test_all_messages(reader_cls: Union[Type[SeekingReader], Type[NonSeekingReader]]):
```

Comment:
This is to satisfy the type checker, which is otherwise unhappy with me constructing the class with a stream argument. The base class constructor does not take a stream.

Reply:
I see - is it possible to keep "unimplemented" version of the same constructor signature on the base class to avoid that type checker issue?

Again none of this is blocking i'm just curious to get to know this library better.

Reply:
I didn't want to do that because it's not a requirement of future McapReader subclasses to take the same constructor arguments - maybe they could take some other input options, not sure. The constructor is not part of the abstract interface.